src/common/transformations/include/ov_ops/fully_connected_quantized_legacy.hpp

```diff
@@ -0,0 +1,52 @@
+// Copyright (C) 2018-2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include "openvino/core/node.hpp"
+#include "openvino/op/op.hpp"
+#include "ov_ops/fully_connected.hpp"
+
+namespace ov {
+namespace op {
+namespace internal {
+
+class TRANSFORMATIONS_API FullyConnectedQuantizedLegacy : public ov::op::internal::FullyConnected {
+public:
+    OPENVINO_OP("FullyConnectedQuantizedLegacy", "gpu_opset");
+
+    FullyConnectedQuantizedLegacy() = default;
+
+    FullyConnectedQuantizedLegacy(const OutputVector& arguments,
+                                  const ov::element::Type output_type = ov::element::undefined);
+
+    FullyConnectedQuantizedLegacy(const ov::Output<Node>& X,
+                                  const ov::Output<Node>& W,
+                                  const ov::Output<Node>& bias,
+                                  const ov::Output<Node>& deq_scales,
+                                  const ov::Output<Node>& deq_zero_points,
+                                  const ov::element::Type output_type = ov::element::undefined);
```

Comment:
Can it be expressed as:
```
    FullyConnectedQuantized(X, W, bias, placeholder, placeholder, placeholder, placeholder, deq_scales, deq_zero_points)
```
?

Reply:
Probably we will have to do this, if we are not going to change current MatcherPass logic.
I mean currently MatcherPass requires a matcher to have the same number of the inputs as a node which is being matched.
If we relax this requirement (maybe not by default, but by providing some configuration option to the MatcherPass), so we can only specify the node inputs we really care about, then it will be possible to have Operations with variable size of the inputs (still strictly ordered, but the optional tail inputs can be omitted, and we would not need to add so many placeholders for the inputs.
But potentially it can bring some other problems.
@itikhono What do you think?